testing/test_browser.py

```diff
@@ -218,3 +218,67 @@ def test_size(browser):
     width, height = browser.size_of('#exact_dimensions')
     assert width == 42
     assert height == 69
+
+
+def test_current_window_handle(browser):
+    assert browser.title == "Test page"
+    assert browser.current_window_handle
+
+
+def test_window_handles(request, browser):
+    assert len(browser.window_handles) == 1
+    handle = browser.new_window(url="http://example.com")
```

Comment:
let's not include actual HTTP requests, and name resolution, in the unit tests.

Use the test_server fixture to just open another window to the testing page, hosted locally.

Reply:
moved to `test_server.url`